mmocr/datasets/preparers/config_generator.py

```diff
@@ -282,6 +308,19 @@ def __init__(
             val_anns=val_anns,
             test_anns=test_anns,
             config_path=config_path)
+        self.dataset_type = self._get_dataset_type()
+
+    def _get_dataset_type(self) -> str:
+        """Get dataset type based on the dataset_name.
+
+        Returns:
+            str: dataset type
+        """
+        if self.dataset_name.endswith('_lmdb'):
+            dataset_type = 'RecogLMDBDataset'
+        else:
+            dataset_type = 'OCRDataset'
```

Comment:
Would it be better to infer the dataset type from the file extension name? Users can therefore gain more flexibility in customizing the data format for different splits. The logic flow is similar to `mmengine.dump`, which decides the file format based on the filename extension.

Reply:
done~